app/lib/account/backend.dart

```diff
@@ -224,6 +236,16 @@ class AccountBackend {
         return updatedUser;
       }
 
+      // We've probably found a non-valid User state, it is better to fail the
+      // authentication. Let's investigate it before allowing the creation of
+      // the new User.
+      if (usersWithEmail.isNotEmpty) {
+        final userIds = usersWithEmail.map((u) => u.userId).join(', ');
+        _logger.severe(
+            'Probably User state violation while looking for ${auth.email}: $userIds');
```

Comment:
I think it's okay to have two users with the same email, as long as they have different `ouathUserId` or one of them is deleted.

Reply:
I think we just continue if this is the case. What are the cases where this can happen?

User account is deleted and new account with the same email is created. Isn't that the only case where this can happen?

If that's not the case, it's likely datastore inconsistency?

Reply:
I am not sure here... On the one hand I'm tempted to throw an error and give up.

Reply:
I'd rather err on the side where we don't allow it first, and maybe allow it on a case-by-case basis.

Reply:
Is there any risk to creating a new `User` entity?

Clearly if emails can be reused and transferred between google accounts (as they can if they are not @gmail.com address), then this can happen, and is perfectly acceptable.

I think creating a new user is fine, that won't give you access to anything, right?

Reply:
Removed.